src/deploy/GoogleDeployer.js

```diff
@@ -32,16 +32,13 @@ class GoogleDeployer extends BaseDeployer {
   }
 
   ready() {
-    if (!this._client || !this._cfg.projectID || !this._cfg.keyFile) {
-      return false;
-    }
-    this._keyFilename = path.resolve(process.cwd(), this._cfg.keyFile);
-    // todo: maybe support async `ready` ?
-    if (fs.existsSync(this._keyFilename)) {
-      return true;
+    if (this._cfg.keyFile) {
+      this._keyFilename = path.resolve(process.cwd(), this._cfg.keyFile);
+      if (!fs.existsSync(this._keyFilename)) {
+        return false;
```

Comment:
well :-) this is the same as before :-)

Reply:
this just looks more complicated, and it does a `fs.existsSync()` in more cases... also, the last `!!this._cfg.keyFile` is kind of weird. 

I suggest to remove the changes here :-)

Reply:
But then we are stuck with #170 


with helix-deploy (latest release):
```bash
$ npx hedy -v --deploy --target google --test=/_status_check/healthcheck.json --pkgVersion=ci999 -l ci --google-key-file ~/.helix-google.json --google-email cloud-function
                                                              s-dev@helix-225321.iam.gserviceaccount.com --google-project-id helix-226321
universal-action-builder v3.14.3
selected targets: Google
--: creating bundle ...
asset dist/helix-services/status@ci999-bundle.js 1.69 MiB [emitted] (name: main)
runtime modules 211 bytes 2 modules
cacheable modules 1.59 MiB
  javascript modules 1.44 MiB 179 modules
  json modules 154 KiB
    modules by path ./node_modules/@adobe/ 7.44 KiB
      ./node_modules/@adobe/helix-status/package.json 2.44 KiB [built] [code generated]
      ./node_modules/@adobe/helix-log/package.json 2.32 KiB [built] [code generated]
      ./node_modules/@adobe/helix-fetch/package.json 2.68 KiB [built] [code generated]
    ./node_modules/require-in-the-middle/package.json 1.87 KiB [built] [code generated]
    ./node_modules/mime-db/db.json 137 KiB [built] [code generated]
    ./node_modules/resolve/lib/core.json 1.76 KiB [built] [code generated]
    ./node_modules/async-hook-jl/package.json 1.35 KiB [built] [code generated]
    ./node_modules/stack-chain/package.json 1.13 KiB [built] [code generated]
    ./node_modules/axios/package.json 2.61 KiB [built] [code generated]
27 modules

WARNING in ./node_modules/epsagon/dist/bundle.js 1:63476-63483
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
 @ ./node_modules/@adobe/helix-epsagon/src/epsagon.js 29:25-43 55:35-53
 @ ./node_modules/@adobe/helix-epsagon/src/index.js 13:16-39
 @ ./node_modules/@adobe/helix-deploy/src/template/index.js 15:20-51

WARNING in ./node_modules/epsagon/dist/bundle.js 1:63596-63603
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
 @ ./node_modules/@adobe/helix-epsagon/src/epsagon.js 29:25-43 55:35-53
 @ ./node_modules/@adobe/helix-epsagon/src/index.js 13:16-39
 @ ./node_modules/@adobe/helix-deploy/src/template/index.js 15:20-51

2 warnings have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

webpack 5.22.0 compiled with 2 warnings in 1462 ms
ok: created bundle dist/helix-services/status@ci999-bundle.js
--: creating zip file ...
 - package.json
 - function.json
 - index.js
 496223 total bytes
ok: created action: dist/helix-services/status@ci999.zip.
--: validating bundle ...
ok: bundle can be loaded and has a main() function.
--: deploying action to Google ...
Unable to deploy Google Cloud function: Could not load the default credentials. Browse to https://cloud.google.com/docs/authentication/getting-started for more information. Error: Could not load the default credentials. Browse to https://cloud.google.com/docs/authentication/getting-started for more information.
    at GoogleAuth.getApplicationDefaultAsync (/Users/lars/Code/helix-status-service/node_modules/google-auth-library/build/src/auth/googleauth.js:173:19)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async GoogleAuth.getClient (/Users/lars/Code/helix-status-service/node_modules/google-auth-library/build/src/auth/googleauth.js:545:17)
    at async GrpcClient._getCredentials (/Users/lars/Code/helix-status-service/node_modules/google-gax/build/src/grpc.js:109:24)
    at async GrpcClient.createStub (/Users/lars/Code/helix-status-service/node_modules/google-gax/build/src/grpc.js:252:23)
error: Google - Could not load the default credentials. Browse to https://cloud.google.com/docs/authentication/getting-started for more information.
error: aborted due to errors during deploy
```

with the proposed patches it is getting deployed.